### PR TITLE
Serve dapp from backend if given a deployed hostname

### PR DESCRIPTION
**Pull request by mikeshultz**
This PR allows us to serve assembled shops from the backend by pointing a known domain at the backend.

This will require some modifications to the nginx ingress to set the `X-Forwarded-Host` header for prod deployments.

Will follow this up later with an update to DNS instructions stuff but want to get in some testing before that happens.

### CODE REVIEW

File: backend/utils/hostCache.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    lastLookupCleanup = now
+  }
+
+  if (typeof hostnames[host] === 'undefined') {
+    log.debug('Cache miss')
+
+    // We don't want to poke the DB too often for static file requests
+    if (!lastLookup[host] || now - lastLookup[host] > LOOKUP_LIMIT) {
+      const shopDom = await ShopDomain.findOne({
+        where: { domain: host },
```
Comment (franckc) [inline backend/utils/hostCache.js:40]: data is tiny for now so performance will be fine. but might as well add an index on the 'domain' column for the future.

**Review by franckc — APPROVED** [review]
LGTM

Do we have any concern about the performance of serving those files from Google NFS disk? I guess if a shop starts driving decent traffic volume we'll upgrade them to CDN so probably not an issue.

**Comment by mikeshultz** [comment]
> Do we have any concern about the performance of serving those files from Google NFS disk?

NFS reads are pretty good.  There's caching at the OS level.  And as you say, if they get heavy they can be moved to the CDN.

File: backend/utils/hostCache.js (hunk 1 above)
Comment (mikeshultz) [reply backend/utils/hostCache.js:40]: Great point, I'll add it to this PR.

**Pull request merged**